Dev 2.0

GigaZiv opened the pull request:
Домашняя работа

GigaZiv committed:
Dev 2.0
Очень много вопросов по техзаданию, почем именно так и не по-другому....

GigaZiv committed:
Dev 2.0
Очень много вопросов по техзаданию, почем именно так и не по-другому....

ArturNurtdinov commented on app/src/main/java/rs/example/playlistmaker/SearchAcitivty.kt:
```diff
@@ -0,0 +1,84 @@
+class SearchActivity : AppCompatActivity() {
+
+    /* Идентификатор */
+    companion object {
```
`companion object` принято объявлять внизу класса согласно Kotlin Coding Convention (https://kotlinlang.org/docs/coding-conventions.html#class-layout)

GigaZiv replied:
Да, согласен

ArturNurtdinov commented on app/src/main/java/rs/example/playlistmaker/SearchAcitivty.kt:
```diff
@@ -0,0 +1,84 @@
+
+    }
+
+    // Я бы конечно сделал по другому, но раз так надо в ТехЗадании, то хозяин барин
```
Если ты хотел обработать `savedInstanceState` в `onCreate` - то это легально)

GigaZiv replied:
Да именно так

ArturNurtdinov commented on app/src/main/java/rs/example/playlistmaker/SettingsActivity.kt:
```diff
@@ -1,20 +1,53 @@
+                val i = Intent(Intent.ACTION_SEND).apply {
+                    type = "text/plain"
+                    putExtra(Intent.EXTRA_SUBJECT, "Курс Android-разработчик")
+                    putExtra(Intent.EXTRA_TEXT, "https://practicum.yandex.ru/android-developer/")
```
Строчку и ссылку нужно вынести в ресурсы приложения и получать через метод `getString()`

ArturNurtdinov commented on app/src/main/res/drawable/ic_box.png:
Кажется это лишняя картинка, лучше удалить

GigaZiv replied:
++

ArturNurtdinov commented on app/src/main/res/layout/activity_search.xml:
```diff
@@ -0,0 +1,97 @@
+        android:text="@string/search" />
+
+
+<!--Не знаю почему так в тех.задании но лучше это сделать с LinearLayout!!!!!    -->
```
Не обязательно, FrameLayout тоже хорошо подходит в данном случае. В любом случае, закоментированный код лучше удалить

GigaZiv replied:
++

GigaZiv committed:
Dev 2.1

Поправлено (торопился просто не проверил):
1. Почта отправляется
...putExtra(Intent.EXTRA_EMAIL, arrayOf(getString(R.string.myEmail)))
...putExtra(Intent.EXTRA_SUBJECT, getString(R.string.themeSupport))
...putExtra(Intent.EXTRA_TEXT, getString(R.string.massageSupport))

2. Ссылка вынесена
...getString(R.string.urlOfPracticum))

GigaZiv committed:
Dev 2.2

Поправлено (торопился просто не проверил):
1. Почта отправляется
...putExtra(Intent.EXTRA_EMAIL, arrayOf(getString(R.string.myEmail)))
...putExtra(Intent.EXTRA_SUBJECT, getString(R.string.themeSupport))
...putExtra(Intent.EXTRA_TEXT, getString(R.string.massageSupport))

2. Ссылка вынесена
...getString(R.string.urlOfPracticum))

Код поправил, картинку удалил

ArturNurtdinov commented on app/src/main/java/rs/example/playlistmaker/SettingsActivity.kt:
```diff
@@ -26,7 +26,7 @@ class SettingsActivity : AppCompatActivity() {
                 val i = Intent(Intent.ACTION_SEND).apply {
                     type = "text/plain"
                     putExtra(Intent.EXTRA_SUBJECT, "Курс Android-разработчик")
```
Эту строчку тоже необходимо вынести в ресурсы

GigaZiv committed:
Dev 2.3
